### DISCUSSION TITLE
Unity WebGL

### DISCUSSION THREAD

**Discussion by JanetGilbert** [deepgram/community#639] [General help]
My company has a product using Unity WebGL, and we are interested in using Deepgram for speech recognition. I built the Unity example for WebGL and get the following error:

![image](https://github.com/deepgram/community/assets/1903245/19ed30d7-aae6-42f7-bf42-9a17021d72b4)

We are using the Estrada library to support the microphone in WebGL as the microphone is not supported by default, but no other alterations to the demo. 

Everything works fine when run in editor. 

```
var headers = new Dictionary<string, string>
{
    { "Authorization", "Token MYTOKEN" }
};

websocket = new WebSocket("wss://api.deepgram.com/v1/listen?encoding=linear16&sample_rate=" + AudioSettings.outputSampleRate.ToString(), headers);
```

(Where MYTOKEN is replaced by my actual token which is confirmed to work)

**Comment by team-deepgram** [comment]
Thanks for asking your question about Deepgram! If you didn't already include it in your post, please be sure to add as much detail as possible so we can assist you efficiently, such as:
- The `request_id` if you have a question about your requests or transcription responses.
- The features you used or the full `api.deepgram.com` URL you sent your request to, including parameters.
- Any code snippets you can share.

**Comment by DamienDeepgram** [comment]
> My company has a product using Unity WebGL, and we are interested in using Deepgram for speech recognition. I built the Unity example for WebGL and get the following error:

The error in the image is pointing to the need for a user to interact with the page to allow microphone recording

![image](https://github.com/deepgram/community/assets/138720050/bd1e0a4c-1202-48cd-b3e7-578fe48ee598)

**Comment by JanetGilbert** [comment]
Thanks for your reply @DamienDeepgram! 

I don't think that's the reason, because I get the same problem in Firefox which does not require a user gesture.

![image](https://github.com/deepgram/community/assets/1903245/c14785e2-e702-41fb-8d72-0a0ec1fb01a2)

I also put in some debugging to make absolutely sure the mic has been given permission and started recording before making a websocket connection to deepgram.

**Reply by brandondines** [reply]
sorry for the bump, was there ever a solution to this problem as I am experiencing the same thing. It doesn't seem to connect on the websocket when using webgl but works fine in editor.
